hatchery/authz.go

```diff
@@ -0,0 +1,221 @@
+package hatchery
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+type AuthzConfig struct {
+	Version float32          `json:"version"`
+	Rules   AuthzVersion_0_1 `json:"rules"`
```

Comment:
I'm not a Go expert, but is there a way you can have a less restrictive structure looking for the version and then depending on the version, expect a more formalized structure? Basically I'm thinking of how we can actually support moving to a new version in the future without rewriting all this.

Reply:
with some hoop jumping i think i was able to get something more flexible working 🤞  i still need to update the unit tests